cscs-checks/prgenv/cuda_fortran_check.py

```diff
@@ -0,0 +1,38 @@
+# Copyright 2016-2020 Swiss National Supercomputing Centre (CSCS/ETH Zurich)
+# ReFrame Project Developers. See the top-level LICENSE file for details.
+#
+# SPDX-License-Identifier: BSD-3-Clause
+
+import reframe as rfm
+import reframe.utility.sanity as sn
+
+
+@rfm.required_version('>=2.14')
+@rfm.simple_test
+class CUDAFortranCheck(rfm.RegressionTest):
+    def __init__(self):
+        self.valid_systems = ['daint:gpu', 'dom:gpu', 'tiger:gpu']
+        self.valid_prog_environs = ['PrgEnv-pgi']
+        self.sourcepath = 'vecAdd_cuda.f90'
+
+        self.num_tasks = 1
+        self.modules = ['craype-accel-nvidia60']
+
+        self.executable = self.name
+        self.build_system = 'SingleSource'
+        self.num_gpus_per_node = 1
+        self.num_tasks_per_node = 1
+        result = sn.extractsingle(r'final result:\s+(?P<result>\d+\.?\d*)',
+                                  self.stdout, 'result', float)
+        self.sanity_patterns = sn.assert_reference(result, 1., -1e-5, 1e-5)
+
+        self.maintainers = ['TM', 'AJ']
+        self.tags = {'production', 'craype'}
+
+    @rfm.run_before('compile')
+    def setflags(self):
```

Comment:
This is the workaround so please name the hook accordingly. Check how other tests are doing it, e.g., `cscs-checks/prgenv/opencl_check.py`.

Reply:
I do not understand this last comment